backend/tracim_backend/lib/webdav/resources.py

```diff
@@ -68,143 +70,224 @@ def wrapper(self: "_DAVResource", *arg, **kwarg) -> typing.Callable:
     return decorator
 
 
-class RootResource(DAVCollection):
+def get_workspace_resource(
+    path: str,
+    environ: typing.Dict,
+    workspace: Workspace,
+    label: str,
+    tracim_context: "WebdavTracimContext",
+) -> "WorkspaceResource":
+    return WorkspaceResource(
+        path=path, environ=environ, workspace=workspace, tracim_context=tracim_context, label=label,
+    )
+
+
+def get_content_resource(
+    content: Content,
+    path: str,
+    environ: typing.Dict,
+    workspace: Workspace,
+    tracim_context: "WebdavTracimContext",
+) -> _DAVResource:
     """
-    RootResource ressource that represents tracim's home, which contains all workspaces
+    Helper to get correct content webdav resource according to content type
     """
+    if content.type == content_type_list.Folder.slug:
+        return FolderResource(
+            path=path,
+            environ=environ,
+            workspace=workspace,
+            content=content,
+            tracim_context=tracim_context,
+        )
+    elif content.type == content_type_list.File.slug:
+        return FileResource(
+            path=path, environ=environ, content=content, tracim_context=tracim_context,
+        )
+    else:
+        return OtherFileResource(
+            path=path, environ=environ, content=content, tracim_context=tracim_context
+        )
 
-    def __init__(self, path: str, environ: dict, tracim_context: "WebdavTracimContext"):
-        super(RootResource, self).__init__(path, environ)
+
+class WebdavContainer(ABC):
+    """
+    A Webdav Container should implement theses methods.
+    """
+
+    @abstractmethod
+    def createEmptyResource(self, file_name: str):
+        pass
+
+    @abstractmethod
+    def createCollection(self, label: str) -> "FolderResource":
+        pass
+
+    @abstractmethod
+    def getMemberNames(self) -> [str]:
+        pass
+
+    @abstractmethod
+    def getMember(self, label: str) -> _DAVResource:
+        pass
+
+    @abstractmethod
+    def getMemberList(self) -> [_DAVResource]:
+        pass
+
+
+class WorkspaceOnlyContainer(WebdavContainer):
+    """
+    Container that can get children workspace
+    """
+
+    def __init__(
+        self,
+        path: str,
+        environ: dict,
+        label: str,
+        workspace: typing.Optional[Workspace],
+        provider: "TracimDavProvider",
+        tracim_context: "WebdavTracimContext",
+        list_orphan_workspaces: bool = False,
+    ) -> None:
+        """
+        Some rules:
+        - if workspace given is None, return workspaces with no parent
+        - if workspace given is correct, return
+        - if list_orphan_workspaces is True, it
+         add user known workspaces without any user known parent to the list.
+         - in case of workspace collision, only the first named workspace (sorted by workspace_id
+         from lower to higher) will be returned
```

Comment:
```suggestion
        - if the given workspace is None, return workspaces with no parent
        - if the given workspace is correct, return [WHAT - feel free to edit my suggestion to replace this]
        - if list_orphan_workspaces is True, it
         adds user-known workspaces without any user-known parent to the list.
         - in case of workspace collision, only the first named workspace (sorted by workspace_id
         from lower to higher) will be returned
```
This list of rules seems similar to another one present in this PR, would it be possible to put these all at a same place and refer to them instead? (maybe it's not a good idea, I don't know)

Reply:
Theses list are differents, i will not mix them because it make things unclear.